nncf/common/graph/graph.py

```diff
@@ -700,3 +702,16 @@ def remove_nodes_from(self, nodes: List[NNCFNode]) -> None:
         self._node_id_to_key_dict = {}
         for node_key, node in self._nx_graph.nodes.items():
             self._node_id_to_key_dict[node["id"]] = node_key
+
+    def find_patterns_nodes(self, patterns: GraphPattern) -> List[NNCFNode]:
```

Comment:
Could be reused in the `get_ip_graph_with_merged_hw_optimized_operations`?

Reply:
I am afraid not. `get_ip_graph_with_merged_hw_optimized_operations` works upon `InsertionPointGraph`, while `find_patterns_nodes` is a method of `NNCFGraph`